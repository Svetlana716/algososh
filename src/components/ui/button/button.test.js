import { Button } from './button';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe ('button test case', () => {
    test ('button with text', () => {
    const tree = renderer
      .create(<Button text="text" />)
      .toJSON();

      expect(tree).toMatchSnapshot();
  });

  test ('button without text', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();

      expect(tree).toMatchSnapshot();
  });

  test ('disabled button', () => {
    const tree = renderer
      .create(<Button disabled />)
      .toJSON();

      expect(tree).toMatchSnapshot();
  });

  test ('button with loader', () => {
    const tree = renderer
      .create(<Button isLoader />)
      .toJSON();

      expect(tree).toMatchSnapshot();
  });

  test ('button click callback', () => {
    const handleClick = jest.fn();
    const { container } = render(<Button onClick={handleClick}/>);
    const button = container.firstChild;
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

});