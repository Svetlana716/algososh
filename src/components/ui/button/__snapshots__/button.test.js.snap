// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test case button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button test case button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;

exports[`button test case button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button test case disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
